File: backend/src/services/aiOllamaService.js
import axios from 'axios';

const MCP_API_TOOL = 'http://mcp_tools:5001/search';
// Gửi prompt đến MCP (Python Server)
export const askLlamaViaMCP = async (prompt) => {
    try {
        const res = await axios.post(MCP_API_TOOL, { prompt });
        return res.data.response;
    } catch (err) {
        console.error('MCP error:', err.message);
        throw err;
    }
};

File: backend/app/services/ai_ollama_service.py
import requests

MCP_API_TOOL = 'http://mcp_tools:5001/search'

def ask_llama_via_mcp(prompt):
    try:
        response = requests.post(MCP_API_TOOL, json={'prompt': prompt})
        response.raise_for_status()
        return response.json().get('response', '')
    except requests.RequestException as e:
        print('MCP error:', str(e))
        raise e
